PXB-2516 : debug build fails on Xenial-32

kennt-percona opened the pull request:
https://jira.percona.com/browse/PXB-2516

Fixing build failures.

kennt-percona commented:
https://pxb.cd.percona.com/view/PXB%208.0/job/percona-xtrabackup-8.0-param/251/

rahulmalik87 commented on storage/innobase/xtrabackup/src/quicklz/quicklz.h:
```diff
@@ -62,7 +62,7 @@ typedef unsigned short int ui16;
 // Detect if pointer size is 64-bit. It's not fatal if some 64-bit target is not
 // detected because this is only for adding an optional 64-bit optimization.
-#if defined _LP64 || defined __LP64__ || defined __64BIT__ || _ADDR64 || \
+#if defined _LP64 || defined __LP64__ || defined __64BIT__ || defined _ADDR64 || \
```
I see same file exists in 2.4? is it impacting build failure for 2.4 as well?

kennt-percona replied:
Interestingly, the compile succeeds on 2.4 (no warnings)

rahulmalik87 replied:
may be some dependency version. I think better to add fix in 2.4 as well

kennt-percona replied:
Fixed in 2.4 also.  https://github.com/percona/percona-xtrabackup/pull/1113

kennt-percona committed:
PXB-2516 : Ubuntu xenial 32 build break

https://jira.percona.com/browse/PXB-2516

Issue
In preprocessor, trying to access a variable that hasn't been defined.
Plus, fixes for server downloads (when downloading a .tar.gz file, perform
the gunzip separately instead of relying on the tar to do the decompression).

kennt-percona committed:
Merge remote-tracking branch 'origin/2.4-pxb-2516' into 8.0-pxb-2516

kennt-percona commented:
https://pxb.cd.percona.com/view/PXB%208.0/job/percona-xtrabackup-8.0-param/252/

rahulmalik87 reviewed (approved):
LGTM
